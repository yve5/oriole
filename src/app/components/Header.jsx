import React from 'react';
import { connect } from 'react-redux';

import * as All from '../actions/All';
import { getScore, getVector } from '../../cvss4';

const Header = ({
  root: {
    cvss4: { metrics },
  },
  resetScore,
}) => {
  const currentScore = getVector(metrics);

  return (
    <header style={{ marginTop: '7rem' }}>
      <div className="container fixed-top bg-body">
        <div className="row">
          <div className="col-lg-8 offset-lg-2 border-bottom">
            <div className="d-flex justify-content-between align-self-center">
              <span className="display-5">CVSS 4.0</span>
              <span className="display-5 text-end">
                Score:{getScore(metrics)}
              </span>
            </div>

            <div className="input-group input-group-sm my-2">
              <button
                className="btn btn-outline-secondary"
                data-testid="button-copy-vector"
                type="button"
                onClick={() =>
                  // eslint-disable-next-line
                  window.prompt(
                    'Copy to clipboard: Ctrl+C, Enter',
                    currentScore
                  )
                }
              >
                Copy
              </button>

              <input
                className="form-control"
                value={currentScore}
                type="text"
                readOnly
              />

              <button
                className="btn btn-outline-danger"
                data-testid="button-reset-score"
                onClick={() => resetScore()}
                type="button"
              >
                Reset
              </button>
            </div>
          </div>
        </div>
      </div>
    </header>
  );
};

const mapStateToProps = (state) => ({
  root: state,
});

const mapDispatchToProps = {
  ...All,
};

export default connect(mapStateToProps, mapDispatchToProps)(Header);
