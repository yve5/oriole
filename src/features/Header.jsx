import React from 'react';
import { Link } from 'react-router-dom';
import Logo from '../assets/logo.svg';

const Header = () => (
  <>
    <header className="sticky-top header-minimized">
      <nav
        className="navbar navbar-dark bg-dark navbar-expand-lg"
        aria-label="Main navigation"
      >
        <div className="container-fluid">
          <div className="navbar-brand me-auto me-lg-4">
            <span className="stretched-link">
              <img
                src={Logo}
                width="50"
                height="50"
                alt="Boosted - Back to Home"
                loading="lazy"
              />
            </span>
          </div>

          <button
            className="navbar-toggler collapsed"
            type="button"
            data-bs-toggle="collapse"
            data-bs-target=".bd-navbar"
            aria-controls="bd-navbar1 bd-navbar2"
            aria-expanded="false"
            aria-label="Toggle navigation"
          >
            <span className="navbar-toggler-icon" />
          </button>

          {/*
           */}
          <div className="navbar-collapse collapse me-lg-auto bd-navbar mb-0 border-0">
            <ul className="navbar-nav">
              <li className="nav-item">
                <span className="nav-link">Home</span>
              </li>
              <li className="nav-item">
                <span className="nav-link active" aria-current="true">
                  Docs
                </span>
              </li>
              <li className="nav-item">
                <span className="nav-link">Examples</span>
              </li>
              <li className="nav-item">
                <span className="nav-link">Guidelines</span>
              </li>
            </ul>
          </div>

          {/*
          <div
            id="bd-navbar2"
            className="navbar-collapse collapse bd-navbar border-0 ms-lg-auto"
          >
            <ul className="navbar-nav flex-row">
              <li className="nav-item">
                <span className="nav-link nav-icon">
                  <svg
                    xmlns="http://www.w3.org/2000/svg"
                    viewBox="0 0 512 499.36"
                    role="img"
                  >
                    <title>GitHub</title>
                    <path
                      fill="currentColor"
                      fillRule="evenodd"
                      d="M256 0C114.64 0 0 114.61 0 256c0 113.09 73.34 209 175.08 242.9 12.8 2.35 17.47-5.56 17.47-12.34 0-6.08-.22-22.18-.35-43.54-71.2 15.49-86.2-34.34-86.2-34.34-11.64-29.57-28.42-37.45-28.42-37.45-23.27-15.84 1.73-15.55 1.73-15.55 25.69 1.81 39.21 26.38 39.21 26.38 22.84 39.12 59.92 27.82 74.5 21.27 2.33-16.54 8.94-27.82 16.25-34.22-56.84-6.43-116.6-28.43-116.6-126.49 0-27.95 10-50.8 26.35-68.69-2.63-6.48-11.42-32.5 2.51-67.75 0 0 21.49-6.88 70.4 26.24a242.65 242.65 0 0 1 128.18 0c48.87-33.13 70.33-26.24 70.33-26.24 14 35.25 5.18 61.27 2.55 67.75 16.41 17.9 26.31 40.75 26.31 68.69 0 98.35-59.85 120-116.88 126.32 9.19 7.9 17.38 23.53 17.38 47.41 0 34.22-.31 61.83-.31 70.23 0 6.85 4.61 14.81 17.6 12.31C438.72 464.97 512 369.08 512 256.02 512 114.62 397.37 0 256 0z"
                    />
                  </svg>
                  <span className="visually-hidden">GitHub</span>
                </span>
              </li>
            </ul>
          </div>
          */}
        </div>
      </nav>
    </header>

    <nav className="navbar navbar-expand-lg navbar-dark bg-dark mb-5">
      <div className="container-fluid">
        {/*
        <div className="navbar-brand">
        <img loading="lazy" alt="Oriole" height="12" width="12" src={Logo} />
        </div>
        */}

        <button className="navbar-toggler" type="button">
          <span className="navbar-toggler-icon" />
        </button>

        <div className="collapse navbar-collapse">
          <ul className="navbar-nav">
            <li className="nav-item">
              <Link className="nav-link" to="/">
                Paging
              </Link>
            </li>
            <li className="nav-item">
              <Link className="nav-link" to="/buttons">
                Buttons
              </Link>
            </li>
          </ul>
        </div>
      </div>
    </nav>
  </>
);

export default Header;
