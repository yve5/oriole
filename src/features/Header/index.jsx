import Logo from '../../assets/logo.svg';

const Header = () => (
  <nav
    className="navbar bg-dark border-bottom border-body mb-3"
    data-bs-theme="dark"
  >
    <div className="container-fluid">
      <span className="navbar-brand">
        <img
          className="d-inline-block align-text-top me-2"
          height="24"
          width="30"
          alt="Logo"
          src={Logo}
        />
        Oriole
      </span>
    </div>
  </nav>
);

export default Header;
