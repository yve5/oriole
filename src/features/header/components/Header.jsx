import React from 'react';
import { clsx } from 'clsx';
import { Link } from 'react-router-dom';

import Logo from '../resources/logo.webp';
import Theme from '../../../shared/theme';

import { Logout } from '../../icons';
import { LangSwitch } from '../../../shared/i18n';

const Header = () => (
  <header>
    <div className={clsx('px-3', 'py-2', 'text-bg-dark', 'border-bottom')}>
      <div
        className={clsx(
          'd-flex',
          'align-items-center',
          'justify-content-between'
        )}
      >
        <Link
          to="/"
          className={clsx(
            'd-flex',
            'align-items-center',
            'my-2',
            'my-sm-0',
            'me-sm-auto',
            'text-white',
            'text-decoration-none'
          )}
        >
          <img
            src={Logo}
            alt="Oriale"
            title="Oriale"
            className={clsx('app-logo')}
          />
        </Link>

        <ul
          className={clsx(
            'nav',
            'col-sm-auto',
            'justify-content-center',
            'text-small',
            'my-2',
            'my-sm-0'
          )}
        >
          <li>
            <LangSwitch
              className={clsx('nav-link', 'text-secondary', 'text-white')}
            />
          </li>
          <li>
            <Theme />
          </li>
          <li>
            <button
              type="button"
              title="Logout"
              className={clsx('nav-link', 'text-secondary', 'text-white')}
            >
              <Logout />
            </button>
          </li>
        </ul>
      </div>
    </div>
  </header>
);

export default Header;
