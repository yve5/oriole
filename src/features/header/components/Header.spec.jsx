import React from 'react';
import { render } from '@testing-library/react';
import { MemoryRouter } from 'react-router-dom';

import Header from './Header';

describe('Header', () => {
  it('renders the component. default', () => {
    const instance = (
      <MemoryRouter>
        <Header />
      </MemoryRouter>
    );

    const { asFragment, rerender } = render(instance);

    rerender(instance);

    expect(asFragment()).toMatchSnapshot();
  });
});
