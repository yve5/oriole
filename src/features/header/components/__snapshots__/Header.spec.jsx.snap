// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Header > renders the component. default 1`] = `
<DocumentFragment>
  <header>
    <div
      class="px-3 py-2 text-bg-dark border-bottom"
    >
      <div
        class="d-flex align-items-center justify-content-between"
      >
        <a
          class="d-flex align-items-center my-2 my-sm-0 me-sm-auto text-white text-decoration-none"
          href="/"
        >
          <img
            alt="Oriale"
            class="app-logo"
            src="/oriole/src/features/header/resources/logo.webp"
            title="Oriale"
          />
        </a>
        <ul
          class="nav col-sm-auto justify-content-center text-small my-2 my-sm-0"
        >
          <li>
            <button
              class="nav-link text-secondary text-white"
              title="FranÃ§ais"
              type="button"
            >
              fr
            </button>
          </li>
          <li>
            <button
              class="nav-link text-secondary text-white"
              data-testid="theme-button"
              title="light"
              type="button"
            >
              <svg
                class="svg-icon svg-icon--md"
                fill="currentColor"
                height="16"
                viewBox="0 -960 960 960"
                width="16"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M480-360q50 0 85-35t35-85q0-50-35-85t-85-35q-50 0-85 35t-35 85q0 50 35 85t85 35Zm0 40q-66.85 0-113.42-46.58Q320-413.15 320-480t46.58-113.42Q413.15-640 480-640t113.42 46.58Q640-546.85 640-480t-46.58 113.42Q546.85-320 480-320ZM80-460q-8.5 0-14.25-5.76T60-480.03q0-8.51 5.75-14.24T80-500h100q8.5 0 14.25 5.76t5.75 14.27q0 8.51-5.75 14.24T180-460H80Zm700 0q-8.5 0-14.25-5.76T760-480.03q0-8.51 5.75-14.24T780-500h100q8.5 0 14.25 5.76t5.75 14.27q0 8.51-5.75 14.24T880-460H780ZM479.97-760q-8.51 0-14.24-5.75T460-780v-100q0-8.5 5.76-14.25t14.27-5.75q8.51 0 14.24 5.75T500-880v100q0 8.5-5.76 14.25T479.97-760Zm0 700q-8.51 0-14.24-5.75T460-80v-100q0-8.5 5.76-14.25t14.27-5.75q8.51 0 14.24 5.75T500-180v100q0 8.5-5.76 14.25T479.97-60ZM254.46-678.77l-57.61-55.85q-5.85-5.61-5.73-13.76.11-8.16 5.96-14.77 6.38-6.62 14.15-6.62 7.77 0 14.15 6.62L282-706.31q6.38 6.62 6.38 14.16 0 7.53-6 14.15t-13.65 6.12q-7.65-.5-14.27-6.89Zm480.16 481.92L678-253.69q-6.38-6.62-6.38-14.27 0-7.66 6.38-14.04 5.23-6.62 13.08-6.12 7.84.5 14.46 6.89l57.61 55.85q5.85 5.61 5.73 13.76-.11 8.16-5.96 14.77-6.38 6.62-14.15 6.62-7.77 0-14.15-6.62ZM678-677.62q-6.62-6-6.12-13.65t6.89-14.27l55.85-57.61q5.61-5.85 13.76-5.73 8.16.11 14.77 5.96 6.62 6.38 6.62 14.15 0 7.77-6.62 14.15L706.31-678q-6.62 6.38-14.16 6.38-7.53 0-14.15-6ZM196.85-196.68q-6.62-6.78-6.62-14.55 0-7.77 6.62-14.15L253.69-282q6.62-6.38 14.27-6.38 7.66 0 14.04 6.38 5.85 5.23 5.35 13.08-.5 7.84-6.12 14.46l-55.85 57.61q-6.38 6.62-14.15 6.5-7.77-.11-14.38-6.33ZM480-480Z"
                  stroke="currentColor"
                  stroke-width="12"
                />
              </svg>
            </button>
          </li>
          <li>
            <button
              class="nav-link text-secondary text-white"
              title="Logout"
              type="button"
            >
              <svg
                class="svg-icon"
                fill="currentColor"
                height="16"
                viewBox="0 -960 960 960"
                width="16"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M224.62-160q-27.62 0-46.12-18.5Q160-197 160-224.62v-510.76q0-27.62 18.5-46.12Q197-800 224.62-800h236.15q8.54 0 14.27 5.73t5.73 14.27q0 8.54-5.73 14.27T460.77-760H224.62q-9.24 0-16.93 7.69-7.69 7.69-7.69 16.93v510.76q0 9.24 7.69 16.93 7.69 7.69 16.93 7.69h236.15q8.54 0 14.27 5.73t5.73 14.27q0 8.54-5.73 14.27T460.77-160H224.62Zm497.76-300H387.69q-8.54 0-14.27-5.73T367.69-480q0-8.54 5.73-14.27t14.27-5.73h334.69l-78.84-78.85q-5.62-5.61-6-13.53-.39-7.93 6-14.54 6.38-6.62 14.15-6.73 7.77-.12 14.39 6.5l104.54 104.53q9.69 9.7 9.69 22.62 0 12.92-9.69 22.62L672.08-352.85q-5.85 5.85-13.89 6.12-8.04.27-14.65-6.35-6.39-6.61-6.27-14.27.11-7.65 6.5-14.03L722.38-460Z"
                  stroke="currentColor"
                  stroke-width="12"
                />
              </svg>
            </button>
          </li>
        </ul>
      </div>
    </div>
  </header>
</DocumentFragment>
`;
